examples/recorder-crx/src/background.ts

```diff
@@ -104,6 +104,13 @@ async function getCrxApp() {
 async function attach(tab: chrome.tabs.Tab, mode?: Mode) {
   if (!tab?.id || (attachedTabIds.has(tab.id) && !mode))
     return;
+   // if the tab is incognito, chek if can be started in incognito mode.
+  if (tab.incognito) {
+    if (!await chrome.extension.isAllowedIncognitoAccess() || !settings.playInIncognito) {
```

Comment:
We cannot have async calls at this point, check line 118 comment (basically sidepanel won't open if it's called after an async call, it must be called synchronously with a user gesture)